csharp/github/crypto.qll

```diff
@@ -0,0 +1,169 @@
+import csharp
+
+module Crypto {
+  class HashingAlgorithm extends DataFlow::ExprNode {
+    abstract DataFlow::ExprNode getHashValue();
+
+    abstract DataFlow::ExprNode getSalt();
+
+    abstract int defaultIterations();
+
+    abstract DataFlow::ExprNode getIterations();
+  }
+
+  class HMacSigningAlgorithm extends DataFlow::ExprNode {
+    abstract string algorithm();
+
+    abstract DataFlow::ExprNode key();
+  }
+
+  class AsymmetricAlgorithm extends DataFlow::ExprNode {
```

Comment:
Should this really inherit from `DataFlow::ExprNode`? This might be a case for an [algebraic datatype](https://codeql.github.com/docs/ql-language-reference/types/#algebraic-datatypes).